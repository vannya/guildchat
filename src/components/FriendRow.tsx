import React from 'react';

export interface IFriendRowProps {
	id: string;
	name: string;
	activeFriend: boolean;
	onClick: (id: string) => void;
}

export const FriendRow = ({
	id,
	name,
	activeFriend,
	onClick
}: IFriendRowProps): JSX.Element => {
	const handleOnClick = (): void => {
		onClick(id);
	};
	return (
		<button
			className={activeFriend ? 'friend-row--active' : 'friend-row'}
			onClick={handleOnClick}
		>
			{name}
		</button>
	);
};
