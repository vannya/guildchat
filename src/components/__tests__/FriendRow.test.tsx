import { cleanup, render } from '@testing-library/react';
import React from 'react';
import userEvents from '@testing-library/user-event';
import { FriendRow, IFriendRowProps } from '../FriendRow';

describe('FriendRow', () => {
	const getProps = (): IFriendRowProps => {
		return {
			id: '1',
			name: 'Test Name',
			activeFriend: false,
			onClick: jest.fn()
		};
	};

	it('should match snapshot', () => {
		const props = getProps();
		const { container } = render(<FriendRow {...props} />);
		expect(container).toMatchSnapshot();
		cleanup();
	});

	it('should handle the button click', () => {
		const props = getProps();
		props.activeFriend = true;
		const { getAllByText } = render(<FriendRow {...props} />);
		const button = getAllByText('Test Name');
		expect(button).toHaveLength(1);
		userEvents.click(button[0]);
		expect(props.onClick).toHaveBeenCalledWith(props.id);
		expect(props.onClick).toHaveBeenCalledTimes(1);
		cleanup();
	});

	it('should display the active className', () => {
		const props = getProps();
		props.activeFriend = true;
		const { getAllByText } = render(<FriendRow {...props} />);
		const button = getAllByText('Test Name');
		expect(button).toHaveLength(1);
		expect(button[0].classList).toContain('friend-row--active');
		cleanup();
	});
});
