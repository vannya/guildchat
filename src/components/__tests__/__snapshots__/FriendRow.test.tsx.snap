// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendRow should match snapshot 1`] = `
<div>
  <button
    class="friend-row"
  >
    Test Name
  </button>
</div>
`;
